### DISCUSSION TITLE
Environment variables for a Shopify Hydrogen deployment

### DISCUSSION THREAD

**Discussion by Moelafi** [vercel/community#935] [Help] [answered]
Hello community,

I am trying to deploy a hydrogen app on Vercel, I used the template peovided on Vercel for Hydrogen apps. Things got confusing when I was trying to add environment variables for the store. 

I the "process" node object is not defined in Hydrogen so cant use that. The standard Vite import meta.env also didnt work. 

Did anyone have this issue before ?

**Comment by amyegan** [comment, marked as answer]
Hi @Moelafi! Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repo if it is public) so I can get a better understanding of what is happening?

**Reply by Moelafi** [reply]
thank you for your response ! sure,

the public repo is: https://github.com/Moelafi/hydrogen
its identical to the Hydrogen deployment template provided by Vercel.

the only thing I changed is in hydrogen.config:

from:
storefrontToken: "...API KEY",

to:
 storefrontToken: process.env.PRIVATE_STOREFRONT_KEY,

after creating a .env and adding the key there it under a variable: PRIVATE_STOREFRONT_KEY

I also added the env variable in the deployment settings in vercel.

after the deployment runs successfully the page load fails due to failure to connect to data source. the API KEY ENV var is not being delivered to the process.

**Reply by amyegan** [reply]
Thanks for sharing this! I found this closed issue that seems related: https://github.com/Shopify/hydrogen/issues/55#issuecomment-978022707

I was able to get my own copy of the template to work by adding `VITE_TOKEN` to my Project's environment variables and using `import.meta.env.VITE_TOKEN` in hydrogen.config.ts. This option is, of course, not suitable for sensitive information.

I also got it to work when I changed `process.env.PRIVATE_STOREFRONT_KEY` to `process.env["PRIVATE_STOREFRONT_KEY"]`, as suggested in https://github.com/vitejs/vite/issues/3176. This second option should work for sensitive information.

**Reply by Moelafi** [reply]
massive thank you ! that fixed the issue for me
